src/android/CodePushReportingManager.java

```diff
@@ -72,6 +46,34 @@ public void run() {
         });
     }
 
+    public boolean hasFailedReport() {
+        if (this.hasFailedReport == null) {
+            this.hasFailedReport = codePushPreferences.getFailedReport() != null;
+        }
+
+        return this.hasFailedReport;
```

Comment:
So it turns out it actually works! Thanks. I stupidly tried it on a void method and that's why it didn't work initially. However, I want to keep the original code cuz its more readable, if that's ok with you.